lib/Backend/NativeCodeGenerator.cpp

```diff
@@ -3629,43 +3629,57 @@ bool NativeCodeGenerator::TryAggressiveInlining(Js::FunctionBody *const topFunct
 void
 JITManager::HandleServerCallResult(HRESULT hr, RemoteCallType callType)
 {
+    // handle the normal hresults
     switch (hr)
     {
     case S_OK:
-        break;
+        return;
     case E_ABORT:
         throw Js::OperationAbortedException();
     case E_OUTOFMEMORY:
         Js::Throw::OutOfMemory();
     case VBSERR_OutOfStack:
         throw Js::StackOverflowException();
     default:
-        // we should not have RPC failure if JIT process is still around
-        if (GetJITManager()->IsServerAlive())
-        {
-            RpcFailure_fatal_error(hr);
-        }
+        break;
+    }
+    // if jit process is terminated, we can handle certain abnormal hresults
 
-        // we only expect to see these hresults in case server has been closed. failfast otherwise
-        if (hr != HRESULT_FROM_WIN32(RPC_S_CALL_FAILED) && hr != HRESULT_FROM_WIN32(RPC_S_CALL_FAILED_DNE))
+    // we should not have RPC failure if JIT process is still around
+    DWORD sleepInterval = 0;
+    uint retries = 0;
```

Comment:
Instead of this retry logic, why not use 250ms (which may or mat not be sufficient depending on the speed of the system, but will certainly decrease the odds) timeout value? WaitForSingleObject will return as soon as the process is signaled (dead). It would only wait until 250ms expires when process stays alive throughout the period.